src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -194,13 +196,19 @@ internal protected CompletionProvider GetProvider(CompletionItem item)
             {
                 lock (_gate)
                 {
-                    _nameToProvider.TryGetValue(name, out provider);
+                    provider = _nameToProvider.GetOrAdd(name, _getProviderByName);
                 }
             }
 
             return provider;
         }
 
+        private CompletionProvider GetProviderByName(string providerName)
+        {
+            var providers = GetProviders(roles: null);
```

Comment:
can a comment be provided explaining why passing null for roles is correct here.  I'm not quite understanding.